src/main/java/net/fabricmc/loom/util/srg/SrgNamedWriter.java

```diff
@@ -40,8 +47,35 @@ public static void writeTo(Logger logger, Path srgFile, MappingTree mappings, St
 
 		try (SrgWriter writer = new SrgWriter(Files.newBufferedWriter(srgFile))) {
 			try (TinyMappingsReader reader = new TinyMappingsReader(mappings, from, to)) {
-				writer.write(reader.read());
+				writer.write(reader.read(MappingSet.create(new ClassesAlwaysHaveDeobfNameFactory())));
```

Comment:
I would check legacyForge here, although this file isn't used with ModLauncher Forge in normal setups, but this is used for legacy support for directly using mixin without using Fabric / Arch's fork, since those read the srg file for remapping the refmaps.